### DISCUSSION TITLE
i18n: How to configure a partially translated site?

### DISCUSSION THREAD

**Discussion by tm-a-t** [vuejs/vitepress#4115] [Q&A] [answered]
Hi! I have a site which is mainly in English but has some pages translated to Russian. I want the Russian navbar and sidebars mirror the English ones, so that the users see there is more content. Is there a good way to set this up?

These are the options I considered:

- Redirecting from `/ru/*` to `/en/*`. Seems not user-friendly, because it means when a user goes from a Russian page to an English page, they will switch to the English version of the site and only move between English pages afterwards. In addition, it can be frustrating if a user opens a page and it suddenly opens in another language.

- Copy the English pages to the Russian directory (maybe with a warning like "This page is only available in English.") This may work, but the problem is `lang="ru"` will remain which can have bad effects such as triggering browser auto-translate. Or is there a way to override `lang` attribute for certain pages in Vitepress?

**Comment by brc-dd** [comment]
Second option can be easily supported via frontmatter/transformPageData 👀

**Reply by tm-a-t** [reply]
Can it? I don't see `lang` setting in frontmatter reference and specifying something like this in frontmatter doesn't help. `PageData` also doesn't have such field.

**Reply by brc-dd** [reply]
I mean it can be supported. It isn't supported at the moment.

**Reply by tm-a-t** [reply]
Can we have this feature please? I can send a PR if needed. It looks like it requires tweaking a single line in `app/data.ts`.

**Comment by tm-a-t** [comment, marked as answer]
Found a better solution.

1. Set the language cookie [(like in Vitepress guide)](https://vitepress.dev/guide/i18n#separate-directory-for-each-locale) but only on the pages that have translation available. 
2. Manually set up redirects by cookie for translated pages.

So even if the user switches between translated and untranslated pages, they always see content in their language if possible.

**Reply by tm-a-t** [reply]
Also, I use CSS to hide the "Russian" button in translation options on the pages without a Russian version

**Reply by cfinnberg** [reply]
Hi, can I ask how do you know that there is no translation available, so you can hide the button?
